src/tags/tests/test_models.py

```diff
@@ -0,0 +1,276 @@
+from django_tenants.test.cases import TenantTestCase
+from django.contrib.auth import get_user_model
+
+from model_bakery import baker
+
+from tags.models import total_xp_by_tags
+from siteconfig.models import SiteConfig
+from quest_manager.models import Quest, QuestSubmission
+from badges.models import Badge, BadgeAssertion
+
+import itertools
+
+User = get_user_model()
+
+
+class Tag_total_xp_by_tags_Tests(TenantTestCase):
+    """ 
+        Specialized TestClass for testing total_xp_by_tags function
+    """ 
+
+    def setUp(self):
+        self.user = baker.make(User)
+
+        # pre determined amount of qs or ba linked to quest/badges
+        self.set_list = [1, 1, 2, 2, 3, 3, 4, 4, 5, 5]  # 10
+        self.set_iter = itertools.cycle(self.set_list)
+
+        # pre determined values for xp
+        self.xp_list = [  # 50
```

Comment:
Please just put the values needed explicitly in each test. I don't think you need more than 5 values for any given test, to avoid slowing down our already slow test suite.

Reply:
done